deno_webgpu/texture.rs

```diff
@@ -0,0 +1,411 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::not_supported;
+use deno_core::error::AnyError;
+use deno_core::ResourceId;
+use deno_core::{OpState, Resource};
+use serde::Deserialize;
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+
+use super::error::WebGpuResult;
+pub(crate) struct WebGpuTexture(pub(crate) wgpu_core::id::TextureId);
+impl Resource for WebGpuTexture {
+    fn name(&self) -> Cow<str> {
+        "webGPUTexture".into()
+    }
+}
+
+pub(crate) struct WebGpuTextureView(pub(crate) wgpu_core::id::TextureViewId);
+impl Resource for WebGpuTextureView {
+    fn name(&self) -> Cow<str> {
+        "webGPUTextureView".into()
+    }
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum GpuTextureFormat {
```

Comment:
Yes. More ideally the `wgpu_types` would just implement `serde::Deserialize` (under a `serde` feature?) on things like `TextureFormat` automatically. That would allow us to remove a bunch of duplicated definitions here, because we could just deserialize JS code to the `wgpu_types` types directly. Have been meaning to open an issue, but haven't had time yet.

Reply:
Ref https://github.com/gfx-rs/wgpu/issues/1887

Reply:
So this can be simplified now?

Reply:
Before: +15,281 −1,974 
After: +14,402 −1,974

Nice!